### ISSUE TITLE
Crash if preconditioner is set to PcNone in 3d

### ISSUE THREAD

**Issue by alexus37** [thunil/mantaflow#4]
I am trying to use the solvePressure function without a Preconditioner (PcNone or 0). 
```python
solvePressure(
                    flags=flags,
                    vel=velocity,
                    pressure=pressure,
                    cgAccuracy=1e-6,
                    cgMaxIterFac=6000 / self.opt.res,
                    retRhs=rhs,
                    preconditioner=0,
                )
```

I think the error is in `source/plugin/pressure.cpp#L407`. I would make a PR to fix it, but only if this repo is active and someone is going to look at it :D

**Comment by sebbas** [comment]
Yes, the assert in `setICPreconditioner()` cannot handle `PC_None`. This should be fixed in the `develop` branch first (not on Github) which then could be merged back into this repositories `master` branch.

**Comment by alexus37** [comment]
okay, I guess then my hands are tied.

**Comment by sebbas** [comment]
Bug reports are still welcome! If you already have a patch you could attach it to this report. It could still be committed on your behalf, just not as a Github PR.
